arbitrum-docs/launch-orbit-chain/how-tos/customize-precompile.mdx

```diff
@@ -395,6 +396,166 @@ Here we can see that the gas cost incurred by the execution of the first transac
+
+We can define a state called `myNumber` here, `myNumber` uses the `storage.StorageBackedUint64` type. You can find more related types in [storage.go][storage_link]:
+
+```
+type ArbosState struct {
+	arbosVersion           uint64                      // version of the ArbOS storage format and semantics
+	upgradeVersion         storage.StorageBackedUint64 // version we're planning to upgrade to, or 0 if not planning to upgrade
+	upgradeTimestamp       storage.StorageBackedUint64 // when to do the planned upgrade
+	networkFeeAccount      storage.StorageBackedAddress
+	l1PricingState         *l1pricing.L1PricingState
+	l2PricingState         *l2pricing.L2PricingState
+	retryableState         *retryables.RetryableState
+	addressTable           *addressTable.AddressTable
+	chainOwners            *addressSet.AddressSet
+	sendMerkle             *merkleAccumulator.MerkleAccumulator
+	blockhashes            *blockhash.Blockhashes
+	chainId                storage.StorageBackedBigInt
+	chainConfig            storage.StorageBackedBytes
+	genesisBlockNum        storage.StorageBackedUint64
+	infraFeeAccount        storage.StorageBackedAddress
+	brotliCompressionLevel storage.StorageBackedUint64 // brotli compression level used for pricing
+	backingStorage         *storage.Storage
+	Burner                 burn.Burner
+	myNumber               storage.StorageBackedUint64 // this is what we added
+}
+```
+
+Next you need to define the offset of your newly added state (note please add it to the end so it won't affect other states):
+
+```
+const (
+	versionOffset Offset = iota
+	upgradeVersionOffset
+	upgradeTimestampOffset
+	networkFeeAccountOffset
+	chainIdOffset
+	genesisBlockNumOffset
+	infraFeeAccountOffset
+	brotliCompressionLevelOffset
+	myNumberOffset // define the offset of your new state here
+)
+```
+
+Then, initialize the state under method [OpenArbosState][arbosstate_openstate] and [InitializeArbosState][arbosstate_initializeArbosState]:
+
+OpenArbosState:
+
+```
+return &ArbosState{
+		arbosVersion,
+		backingStorage.OpenStorageBackedUint64(uint64(upgradeVersionOffset)),
+		backingStorage.OpenStorageBackedUint64(uint64(upgradeTimestampOffset)),
+		backingStorage.OpenStorageBackedAddress(uint64(networkFeeAccountOffset)),
+		l1pricing.OpenL1PricingState(backingStorage.OpenSubStorage(l1PricingSubspace)),
+		l2pricing.OpenL2PricingState(backingStorage.OpenSubStorage(l2PricingSubspace)),
+		retryables.OpenRetryableState(backingStorage.OpenSubStorage(retryablesSubspace), stateDB),
+		addressTable.Open(backingStorage.OpenSubStorage(addressTableSubspace)),
+		addressSet.OpenAddressSet(backingStorage.OpenSubStorage(chainOwnerSubspace)),
+		merkleAccumulator.OpenMerkleAccumulator(backingStorage.OpenSubStorage(sendMerkleSubspace)),
+		blockhash.OpenBlockhashes(backingStorage.OpenSubStorage(blockhashesSubspace)),
+		backingStorage.OpenStorageBackedBigInt(uint64(chainIdOffset)),
```

Comment:
Qq - thoughts on eliminating all but the most relevant lines of code from snippets, to make it easy for devs to identify, copy, and paste the bits they need? (possibly using `//...` syntax to indicate surrounding code)

Reply:
Sure, should we also do this at other code blocks?

Reply:
Yeah that would be awesome! This isn't blocking feedback, just a nice pattern to apply wherever it feels appropriate (can be applied incrementally of course) ^^